app/controllers/projects_controller.rb

```diff
@@ -197,7 +197,7 @@ def set_ivars_for_show
 
   def upload_image_if_present
     # Check if we need to upload an image.
-    upload = params[:project][:upload_image]
+    upload = params[:upload][:image]
```

Comment:
Maybe also `return unless params[:upload]` here, as you did above?

Reply:
Or use `upload = params.dig(:upload, :image)` which is effectively the same thing

Reply:
Switched to dig for both cases.